test/safe.test.ts

```diff
@@ -0,0 +1,127 @@
+  async function createTestWallet(
+    singletonAddress: string,
+    proxyFactory: GnosisSafeProxyFactory & {
+      deploymentTransaction(): ContractTransactionResponse;
+    },
+    safeSingleton: GnosisSafe & {
+      deploymentTransaction(): ContractTransactionResponse;
+    },
+    signers: string[]
+  ) {
+    const initializer = safeSingleton.interface.encodeFunctionData("setup", [
+      signers,
+      2,
+      ethers.ZeroAddress,
+      "0x",
+      ethers.ZeroAddress,
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ]);
+
+    const tx = await proxyFactory.createProxy(singletonAddress, initializer);
+    const receipt = await tx.wait();
+
+    if (!receipt) {
+      throw new Error("No receipt");
+    }
+    const proxyAddress = receipt.logs
+      .map((log) => proxyFactory.interface.parseLog(log))
+      .find((event) => event?.name === "ProxyCreation")?.args.proxy as string;
+
+    const testSafeWallet = await ethers.getContractAt(
+      "GnosisSafe",
+      proxyAddress
+    );
+
+    return testSafeWallet;
+  }
+
+  it("should create a Safe wallet with two signers", async function () {
+    const { signer1, signer2, safeSingleton, proxyFactory } =
+      await safeInitialize();
+
+    const singletonAddress = await safeSingleton.getAddress();
+
+    const testSafeWallet = await createTestWallet(
+      singletonAddress,
+      proxyFactory,
+      safeSingleton,
+      [signer1.address, signer2.address]
+    );
+
+    const owners = await testSafeWallet.getOwners();
+    expect(owners.length).to.equal(2);
+    expect(owners[0]).to.equal(signer1.address);
+    expect(owners[1]).to.equal(signer2.address);
+  });
+
+  describe("LBC Ownership change", function async() {
+    it("should change the ownership of LBC", async function () {
+      const { signer1, signer2, safeSingleton, proxyFactory } =
+        await safeInitialize();
+
+      const singletonAddress = await safeSingleton.getAddress();
+
+      const testSafeWallet = await createTestWallet(
+        singletonAddress,
+        proxyFactory,
+        safeSingleton,
+        [signer1.address, signer2.address]
+      );
+
+      const deployed = await deployLbcProxy(hre.network.name, {
+        verbose: false,
+      });
+      const proxyAddress = deployed.address;
+      const lbc = await ethers.getContractAt(
+        "LiquidityBridgeContract",
+        proxyAddress
+      );
+      const result = await lbc.queryFilter(lbc.getEvent("Initialized"));
+      expect(deployed.deployed).to.be.eq(true);
+      expect(result.length).length.equal(1);
+      expect(await lbc.owner()).to.equal(signer1.address);
+
+      const tx = await lbc.register(...REGISTER_LP_PARAMS);
+      await tx.wait();
+
+      const safeAddress = await testSafeWallet.getAddress();
+      multisigOwners.hardhat.owners = [signer1.address, signer2.address];
+
+      await changeMultisigOwner(safeAddress);
+
+      await expect(
+        lbc.connect(signer1).setProviderStatus(0, true)
```

Comment:
The 1st provider is 1, not 0. I also suggest to run again this function but with the multisig to ensure it works

Reply:
In fact, I ran this test locally with the correct provider ID and its failing because you're registering using the signer 1 and then trying to modify the status as the signer 1, in this case that account was both the owner and the provider so even if it isn't the owner anymore the assertion will fail because it is the provider. My advise is registering with the signer 2 as LP and add the extra assertion I mentioned to ensure it doesn't work with the old owner and works with the new one

<img width="819" alt="image" src="https://github.com/user-attachments/assets/11ff45d5-63a7-4ed2-ade2-f7d9a457b79e" />

Reply:
Solved